ClinGen/clincoded
# Connects to #621. Connects to #1176. Connects to #1181. Connects to #1214. Connects to #1171. Connects to #1213. Connects to #1172. Reduce technical debt, reduce build load time, add users, fix gdm limit, allow for more testing

#  Reduce technical debt


## #1181 upgrade to node 6
 - our front end is entirely JS. but the node package we use was deprecated last summer. node 7 is available but will be only supported for a few months

#### #1181 testing
 - developers should follow the README instructions (updated here to see how to upgrade node locally)

Install Node v6 (v6 is the current LTS version https://github.com/nodejs/LTS#lts-schedule ):

  * check node version::

    $ node --version

  * install via homebrew (homebrew will indicate if you need to unlink a prior version of node)::

    $ brew install node@6

  * or install via nvm::

    $ npm install -g nvm
    $ nvm install 6
    $ nvm use 6


## #621 npm modules
 - tons of old node modules meant fixing a bug was more precarious

 check node module technical debt via `npm outdated`

Before (14 packages need attention):
![image](https://cloud.githubusercontent.com/assets/1878194/21921399/bbb29230-d91c-11e6-8a51-73827a6f370b.png)


#### #621  testing
 - `rm -rf node_modules`
 - `npm install`
 -  check node module status via `npm outdated`

After proposed updates (only 4 need attention):
![image](https://cloud.githubusercontent.com/assets/1878194/21921414/d947f664-d91c-11e6-8803-7eab417ce64d.png)

The most noticeable is the React upgrade which requires re-factoring of some front end components

 - build and start the application as usual
 - run the tests: `bin/test -m bdd -v -v`
 - run the jest tests: `npm test`

## #1176 old encode dependency
 - this is primarily to facilitate the developer experience as buildout runs much faster if you don't have to curl these 2 files every time: `ontology.json` (now saved to repo as it is not updated ever), and remove `backfill_2683_md5sum_content_md5sum.json`

#### #1176  testing
 - `bin/buildout -c buildout-dev.cfg;` does not not curl either file from the encoded-build s3 bucket


# More testing

## #1214 allow login during BDD testing
 - configured testing to use predefined port which are also set at Auth0

#### #1214 testing
 -  Run BDD tests: `bin/test -m bdd -v -v` : port 
 - Chrome should come up with a port within a defined range (65525-65534)
 - Note: this should be refactored in the future so that login happens before all bdd tests. They currently go in any order so if you are logging for one but not logging out it is a problem (i.e. a better test fixture would be better)
 

# Bugs
## #1213 All GDMs page/All Interpretations limited to 25 records
- Currently the limit on some queries to elasticsearch return only 25 results. This meant that people couldn't see all the records on the GDMs page.
- The limit has been raised to a very high number for gdms and interpretations to get around this

#### #1213 testing
- This instance has a copy of the production database so you can't use the Demo Login

- GDMs
https://1176-kd-check-gdm-interp-return-2.demo.clinicalgenome.org/gdm/
You should see >25  GDMs

- Interpretations
https://1176-kd-check-gdm-interp-return-2.demo.clinicalgenome.org/interpretations/
You should see all Interpretations

## #1171 decrease Travis/Sauce sporadic failures
 - sometimes the  Travis BDD tests (the Chrome tests) fail for no reason but work when restarted
 - this makes the waits longer before a timeout and updates the sauce connect package that forms a tunnel to Sauce from Travis. It is hard to reproduce the problem, and therefore hard to test. But hopefully this will help. The problem appears to be related to starting the tests before the tunnel is ready
 - Note: hard to test since it is a sporadic failure

# Users
## #1172 New Users
- Adds many test curators 
- Includes db dumps
